backend/src/apiserver/resource/resource_manager.go

```diff
@@ -261,17 +263,27 @@ func (r *ResourceManager) UpdatePipelineDefaultVersion(pipelineId string, versio
 }
 
 func (r *ResourceManager) CreatePipeline(name string, description string, namespace string, pipelineFile []byte) (*model.Pipeline, error) {
-	// Extract the parameter from the pipeline
-	wf, err := util.ValidateWorkflow(pipelineFile)
-	if err != nil {
-		return nil, util.Wrap(err, "Create pipeline failed")
-	}
-	if wf.IsV2() {
-		overrideV2PipelineName(wf, name, namespace)
-	}
-	paramsJson, err := util.MarshalParameters(wf.Spec.Arguments.Parameters)
-	if err != nil {
-		return nil, util.Wrap(err, "Create pipeline failed")
+	specFormat := util.InferSpecFormat(pipelineFile)
+	var paramsJson string
+	switch specFormat {
+	case model.PipelineSpecFormatV1Argo:
```

Comment:
Instead of using case here, can we abstract the logic into each PipelineSpecFormat using interface?

Reply:
This is a very good suggestion! Yes, I think it's worth making the abstraction in this PR.